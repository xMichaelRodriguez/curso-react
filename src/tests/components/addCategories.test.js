import React from "react";

const { shallow } = require("enzyme");
const { AddCategories } = require("../../components/addCategories");

describe("Pruebas en <AddCategories />", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategories setCategories={setCategories} />);
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategories setCategories={setCategories} />);
  });
  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Dede de cambiar el input", () => {
    const input = wrapper.find("input");
    const value = "hola que hace";
    input.simulate("change", { target: { value } });
  });

  test("No debe de postear la informacion ", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Debe de llamar setcategories y limpiar el input", () => {
    const input = wrapper.find("input");
    const value = "hola que hace2";
    input.simulate("change", { target: { value } });

    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(2);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    expect(input.prop("value")).toBe("");
  });
});
