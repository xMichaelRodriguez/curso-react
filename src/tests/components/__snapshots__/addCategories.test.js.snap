// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategories /> debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="input-field col s12"
    >
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        forof="text"
      >
        Name of gift
      </label>
    </div>
  </div>
</form>
`;
